app/scripts/proactive/view/utils/undo.js

```diff
@@ -37,17 +37,21 @@ define(
                 activeTask: StudioApp.views.workflowView.getActiveTask()};
             this._saveIfDifferent(state);
         },
-        undo: function () {
+        undo: function (num) {
             if (undoStates.length <= 1) {
                 StudioClient.alert("No further undo data", "");
                 return;
             }
-            this._move(undoStates, redoStates);
-            this._restoreLastState();
+            var length = undoStates.length;
+            for(var i= 1; i<num + 1; i++){
+            undoStates = undoStates.slice(0,length - i + 1);
+                this._move(undoStates, redoStates);
+                this._restoreLastState();
+            }
         },
-        undoIfEnabled: function () {
+        undoIfEnabled: function (numTasks) {
             if (enabled) {
-                this.undo();
+                this.undo(numTasks);
```

Comment:
It seems that you narrowed down the use of this function for undoing tasks creation. It can be used for other things, like removing a newly created variable for example.